commons-logs/src/main/java/eu/europeana/api/commons/logs/LoggableDispatcherServlet.java

```diff
@@ -0,0 +1,213 @@
+        // Gorouter Time : is the total time it takes for the request to go through the
+        // Gorouter initially plus the time it takes for the response to travel back through the Gorouter. This does not include
+        // the time the request spends traversing the network to the app. This also does not include the time the
+        // app spends forming a response.
+        //logMessage.setGorouterTime(goRouterTime);
+
+        logMessage.setHttpVersion(StringUtils.substringAfter(requestToCache.getProtocol(), "/"));
+        logMessage.setReferer(requestToCache.getHeader(LogConstants.REFERER));
+        logMessage.setHttpStatus(responseToCache.getStatus());
+        logMessage.setProcessTime(processTime);
+        // server date and time
+        logMessage.setServerDate(StringUtils.substringBefore(serverTime, "T"));
+        logMessage.setServerTime(StringUtils.substringBetween(serverTime, "T", "Z"));
+        logMessage.setServerTimeZoneOffset(LogConstants.SERVER_TIMEZONE_OFFSET);
+
+        logMessage.setUrlQuery(requestToCache.getQueryString());
+        logMessage.setUrlPath(requestToCache.getPathInfo());
+        logMessage.setUserAgent(requestToCache.getHeader(LogConstants.USER_AGENT));
+        logMessage.setVcapRequestId(requestToCache.getHeader(LogConstants.X_VCAP_REQUEST_ID));
+        logMessage.setMethod(requestToCache.getMethod());
+        logMessage.setxB3parentSpanId(requestToCache.getHeader(LogConstants.X_B3_PARENT_SPAN_ID));
+        logMessage.setxB3SpanId(requestToCache.getHeader(LogConstants.X_B3_SPAN_ID));
+        logMessage.setxB3TraceId(requestToCache.getHeader(LogConstants.X_B3_TRACE_ID));
+        logMessage.setxGlobalTransId(requestToCache.getHeader(LogConstants.X_GLOBAL_TRANSACTION_ID));
+        logMessage.setCfConnectingIp(requestToCache.getHeader(LogConstants.CF_CONNECTING_IP));
+        logMessage.setCfIpCountry(requestToCache.getHeader(LogConstants.CF_IP_COUNTRY));
+
+        LOG.info(logMessage);
+    }
+
+    /**
+     * Gets the current date and time in yyyy-MM-dd'T'HH:mm:ss.SSS'Z' format
+     *
+     * @return date in yyyy-MM-dd'T'HH:mm:ss.SSS'Z' format
+     */
+    private String getCurrentDateTime() {
+        DateFormat df = new SimpleDateFormat(LogConstants.DATE_FORMAT, Locale.US);
+        df.setTimeZone(TimeZone.getTimeZone(LogConstants.TIME_ZONE));
+        return df.format(new Date());
+    }
+
+    /**
+     * Gets the client's Ip first from X-FORWARDED-FOR (format : client Ip, proxy, proxy)
+     * if null get the remote Address
+     *
+     * @param request
+     * @return client's IP
+     */
+    private static String getClientIP(HttpServletRequest request) {
+        String ipAddress = request.getHeader(LogConstants.X_FORWARDED_FOR);
+        if (StringUtils.isNotEmpty(ipAddress)) {
+            return ipAddress.contains(",") ? ipAddress.split(",")[0] : ipAddress;
+        } else {
+            return request.getRemoteAddr();
+        }
```

Comment:
No need to parse, just copy in it's entirety

Reply:
okay

Reply:
we might want to parse it if we want the client Ip address